packages/internal-workflow-common/src/workflow-options.ts

```diff
@@ -47,18 +47,18 @@ export interface BaseWorkflowOptions {
   cronSchedule?: string;
 
   /**
-   * Specifies additional non-indexed information in result of list workflow. The type of value
-   * can be any object that are serializable by `DataConverter`.
+   * Specifies additional non-indexed information to attach to the Workflow Execution. The values can be anything that
+   * is serializable by {@link DataConverter}.
    */
   memo?: Record<string, any>;
 
   /**
-   * Specifies additional indexed information in result of list workflow. The type of value should
-   * be a primitive (e.g. string, number, boolean). For dates, use Date.toISOString()
+   * Specifies additional indexed information to attach to the Workflow Execution. More info:
+   * https://docs.temporal.io/docs/typescript/search-attributes
    *
-   * Values are always converted using {@link defaultPayloadConverter}, even when a custom data converter is provided.
+   * Values are always converted using {@link JsonPayloadConverter}, even when a custom data converter is provided.
    */
-  searchAttributes?: Record<string, string | number | boolean>;
+  searchAttributes?: Record<string, SearchAttributeValue>;
```

Comment:
If you take my suggestion above this would be:
```suggestion
  searchAttributes?: Record<string, SearchAttributeValue[]>;
```